src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7973,59 +8033,33 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
                     LookupOptions.Default,
                     originalBinder: this,
                     diagnose: false,
-                    ref discardedUseSiteInfo);
+                    ref useSiteInfo);
 
                 if (lookupResult.IsMultiViable)
                 {
                     foreach (var candidate in lookupResult.Symbols)
                     {
                         if (!candidate.IsStatic &&
-                            IsAccessible(candidate, ref discardedUseSiteInfo) &&
+                            IsAccessible(candidate, ref useSiteInfo) &&
                             candidate is MethodSymbol method &&
                             method.OriginalDefinition is var original &&
                             original.ParameterCount == 2 &&
                             isIntNotByRef(original.Parameters[0]) &&
                             isIntNotByRef(original.Parameters[1]))
                         {
-                            CheckImplicitThisCopyInReadOnlyMember(receiverOpt, lengthOrCountProperty.GetMethod, diagnostics);
+                            patternSymbol = method;
                             CheckImplicitThisCopyInReadOnlyMember(receiverOpt, method, diagnostics);
-                            patternIndexerAccess = new BoundIndexOrRangePatternIndexerAccess(
-                                syntax,
-                                receiverOpt,
-                                lengthOrCountProperty,
-                                method,
-                                BindToNaturalType(argument, diagnostics),
-                                method.ReturnType);
                             checkWellKnown(WellKnownMember.System_Range__get_Start);
                             checkWellKnown(WellKnownMember.System_Range__get_End);
+                            checkWellKnown(WellKnownMember.System_Index__GetOffset);
+                            success = true;
                             break;
                         }
                     }
                 }
             }
 
-            cleanup(lookupResult);
-            if (patternIndexerAccess is null)
-            {
-                return false;
-            }
-
-            _ = MessageID.IDS_FeatureIndexOperator.CheckFeatureAvailability(diagnostics, syntax);
-            checkWellKnown(WellKnownMember.System_Index__GetOffset);
-            if (arguments.Names.Count > 0)
-            {
-                diagnostics.Add(
-                    argIsRange
-                        ? ErrorCode.ERR_ImplicitRangeIndexerWithName
-                        : ErrorCode.ERR_ImplicitIndexIndexerWithName,
-                    arguments.Names[0].GetLocation());
-            }
-            return true;
-
-            static void cleanup(LookupResult lookupResult)
-            {
-                lookupResult.Free();
-            }
+            return success;
 
             static bool isIntNotByRef(ParameterSymbol param)
```

Comment:
Is this used anymore?

Reply:
Yes, I extracted three methods from the original.